src/main/java/org/myrobotlab/framework/Service.java

```diff
@@ -175,10 +175,11 @@
   // FIXME SecurityProvider
   protected AuthorizationProvider authProvider = null;
 
-  private Status lastError = null;
-  private Status lastStatus = null;
-  long lastStatusTs = 0;
-  long statusBroadcastLimitMs = 1000;
+  protected Status lastError = null;

```

Comment:
this is complex enough that we might want to pull it out into a small little ServiceStatus class, for the "last*" variables that you have here.